.github/workflows/pypi-cd.yml

```diff
@@ -140,44 +145,51 @@ jobs:
                   # Log the edited .gitignore file
                   cat .gitignore
 
-            - name: Build Python wheels (linux)
-              if: startsWith(matrix.build.NAMED_OS, 'linux')
+            - name: Build Python wheels (linux x86_64)
+              if: matrix.build.TARGET == 'x86_64-unknown-linux-gnu'
               uses: PyO3/maturin-action@v1
               with:
-                  working-directory: ./python
-                  target: ${{ matrix.build.TARGET }}
-                  args: --release --strip --out wheels -i ${{ github.event_name != 'pull_request' && 'python3.8 python3.9 python3.10 python3.11 python3.12 python3.13' || 'python3.12' }}
-                  manylinux: auto
-                  container: ${{ matrix.build.CONTAINER != '' && matrix.build.CONTAINER || '2014' }}
-                  before-script-linux: |
-                      # Install protobuf compiler
-                      if [[ $(`which apt`) != '' ]]
-                      then
-                        echo "installing unzip and curl"
-                        apt install unzip curl -y
-                      fi
-                      PB_REL="https://github.com/protocolbuffers/protobuf/releases"
-                      ARCH=`uname -p`
-                      if [[ $ARCH == 'x86_64' ]]; then 
-                        PROTOC_ARCH="x86_64"
-                      elif [[ $ARCH == 'aarch64' ]]; then 
-                        PROTOC_ARCH="aarch_64"
-                      else 
-                        echo "Running on unsupported architecture: $ARCH. Expected one of: ['x86_64', 'aarch64']"
-                        exit 1 
-                      fi
-                      curl -LO $PB_REL/download/v3.20.3/protoc-3.20.3-linux-${PROTOC_ARCH}.zip
-                      unzip protoc-3.20.3-linux-${PROTOC_ARCH}.zip -d $HOME/.local
-                      export PATH="$PATH:$HOME/.local/bin"
+                working-directory: ./python
+                target: ${{ matrix.build.TARGET }}
+                args: --release --strip --out wheels -i ${{ github.event_name != 'pull_request' && 'python3.9 python3.10 python3.11 python3.12 python3.13' || 'python3.12' }}
+                manylinux: auto
+                container: ${{ matrix.build.CONTAINER }}
+                before-script-linux: |
+                  # Install protobuf compiler
+                  apt update && apt install unzip curl -y
+                  PB_REL="https://github.com/protocolbuffers/protobuf/releases"
+                  curl -LO $PB_REL/download/v3.20.3/protoc-3.20.3-linux-x86_64.zip
+                  unzip protoc-3.20.3-linux-x86_64.zip -d $HOME/.local
+                  export PATH="$PATH:$HOME/.local/bin"
```

Comment:
Do you want to create a script and call it there? It could be stored in `utils` dir for example.